proposals/0000-extract-runtime.md

```diff
@@ -0,0 +1,421 @@
+---
+title: Extraction of runtime layer
+author: Robert Pasiński <robert.pasinski@callstack.com>
+date: 2024-09-07
+---
+
+> **NOTE**: This is an early draft of the RFC. As it affects most of the RN, it is intended to
+> be a bit more generic and to collect feedback and collectively build a good adoption strategy and detailed approach. 
+
+# RFC0000: Extraction of Runtime layer
+
+## Summary
+
+Main goal of this RFC is to extract a new layer in React Native stack from already existing, shared code, 
+that lives between `jsi` and React Native, reducing redundancy.
+
+Shortly, the goals are:
+
+ - Extract the React Native non-ui logic into separate package/layer
+ - Keep compatibility with existing code as much as possible
+ - Make long term maintenance, versioning and updating easier
+ - Make React Native more inclusive towards OOT (out of tree) platforms
+ - Allow faster iterations on headless APIs compatibility (e.g. Web)
+
+> OOT (out of tree platforms) refer to platforms that are not part of the main repo.
+
+## Motivation
```

Comment:
The proposal is missing some core parts specified in the template like "##Basic example" and it includes the parts that are not specified (##Changelog, ##Future opportunities). Let's stick to the template

Reply:
Looking again at the template `Basic example` is optional. I omitted it as there are no code examples that can highlight API changes

Reply:
Basic example could include the usage of react-native-runtime by an app or a library to get a sense of the public API